idaes/models_extra/power_generation/unit_models/cross_flow_heat_exchanger_1D.py

```diff
@@ -0,0 +1,808 @@
+        @self.Constraint(
+            self.flowsheet().config.time,
+            shell.length_domain,
+            doc="shell side wall temperature",
+        )
+        def temp_wall_shell_eqn(b, t, x):
+            return (
+                b.hconv_shell_total[t, x]
+                * (shell.properties[t, x].temperature - b.temp_wall_shell[t, x])
+                * (b.thickness_tube / 2 / b.therm_cond_wall + b.rfouling_shell)
+                == b.temp_wall_shell[t, x] - b.temp_wall_center[t, x]
+            )
+
+        # Center point wall temperature based on energy balance for tube wall heat holdup
+        @self.Constraint(
+            self.flowsheet().config.time,
+            tube.length_domain,
+            doc="center point wall temperature",
+        )
+        def temp_wall_center_eqn(b, t, x):
+            # heat_shell and heat_tube are positive when heat flows into those
+            # control volumes (and out of the wall), hence the negative sign
+            # on heat_accumulation_term
+            return -heat_accumulation_term(b, t, x) == (
+                b.heat_shell[t, x]
+                * b.length_flow_shell
+                / pyunits.convert(b.length_flow_tube, to_units=shell_units["length"])
+                + pyunits.convert(
+                    b.heat_tube[t, x],
+                    to_units=shell_units["power"] / shell_units["length"],
+                )
+            )
+
+        if not self.config.dynamic:
+            z0 = shell.length_domain.first()
+            z1 = shell.length_domain.last()
+
+            @self.Expression(self.flowsheet().config.time)
+            def total_heat_duty(b, t):
+                enth_in = b.hot_side.properties[t, z0].get_enthalpy_flow_terms(p_hot)
+                enth_out = b.hot_side.properties[t, z1].get_enthalpy_flow_terms(p_hot)
+
+                return pyunits.convert(
+                    enth_in - enth_out,  # Hot side loses enthalpy
+                    to_units=shell_units["power"],  # Hot side isn't always the shell
+                )
+
+            @self.Expression(self.flowsheet().config.time)
+            def log_mean_delta_temperature(b, t):
+                dT0 = b.hot_side.properties[t, z0].temperature - pyunits.convert(
+                    b.cold_side.properties[t, z0].temperature,
+                    to_units=shell_units["temperature"],
+                )
+                dT1 = b.hot_side.properties[t, z1].temperature - pyunits.convert(
+                    b.cold_side.properties[t, z1].temperature,
+                    to_units=shell_units["temperature"],
+                )
+                return (dT0 - dT1) / log(dT0 / dT1)
+
+            @self.Expression(self.flowsheet().config.time)
+            def overall_heat_transfer_coefficient(b, t):
+                return b.total_heat_duty[t] / (
+                    b.total_heat_transfer_area * b.log_mean_delta_temperature[t]
+                )
+
+    def initialize_build(
```

Comment:
General comment: I am inclined to not include this method in new models and just provide `Initializer` objects instead, as this API has been deprecated.

Reply:
Is there a reference for this new API?

![image](https://github.com/IDAES/idaes-pse/assets/88728506/bff8c96a-e692-4f27-9e04-bc91f175d5ab)

Reply:
https://idaes-pse.readthedocs.io/en/stable/reference_guides/initialization/index.html